src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -132,7 +132,7 @@ public class BlockPuller : IBlockPuller, IDisposable
         private readonly CancellationTokenSource cancellationSource;
 
         /// <summary>The average block size in bytes calculated used up to <see cref="AverageBlockSizeSamplesCount"/> most recent samples.</summary>
-        /// <remarks>This object has to be protected by <see cref="queueLock" />.</remarks>
+        /// <remarks>Write access to this object has to be protected by <see cref="queueLock" />.</remarks>
```

Comment:
Why write access only?

Reply:
Yeah. For read access it's ok to use it without a lock.